flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/ForStWriteBatchOperation.java

```diff
@@ -0,0 +1,96 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.state.forst;
+
+import org.rocksdb.RocksDB;
+import org.rocksdb.RocksDBException;
+import org.rocksdb.WriteBatch;
+import org.rocksdb.WriteOptions;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * The writeBatch operation implementation for ForStDB.
+ *
+ * @param <K> The type of key in put access request.
+ * @param <V> The type of value in put access request.
+ */
+public class ForStWriteBatchOperation<K, V> implements ForStDBOperation<Void> {
+
+    private static final int PER_RECORD_ESTIMATE_BYTES = 100;
+
+    private final RocksDB db;
+
+    private final List<Request<K, V>> batchRequest;
+
+    private final WriteOptions writeOptions;
+
+    ForStWriteBatchOperation(
+            RocksDB db, List<Request<K, V>> batchRequest, WriteOptions writeOptions) {
+        this.db = db;
+        this.batchRequest = batchRequest;
+        this.writeOptions = writeOptions;
+    }
+
+    @Override
+    public CompletableFuture<Void> process() throws IOException {
+        CompletableFuture<Void> result = new CompletableFuture<>();
+        try (WriteBatch writeBatch =
+                new WriteBatch(batchRequest.size() * PER_RECORD_ESTIMATE_BYTES)) {
+            for (Request<K, V> request : batchRequest) {
+                ForStInnerTable<K, V> table = request.table;
+                if (request.value == null) {
+                    // put(key, null) == delete(key)
+                    writeBatch.delete(
+                            table.getColumnFamilyHandle(), table.serializeKey(request.key));
+                } else {
+                    writeBatch.put(
+                            table.getColumnFamilyHandle(),
+                            table.serializeKey(request.key),
+                            table.serializeValue(request.value));
+                }
+            }
+            db.write(writeOptions, writeBatch);
+            result.complete(null);
+        } catch (RocksDBException e) {
+            result.completeExceptionally(e);
+            throw new IOException("Error while adding data to ForStDB", e);
+        }
+        return result;
+    }
+
+    /** The Put access request for ForStDB. */
+    static class Request<K, V> {
+        final K key;
+        final V value;
+        final ForStInnerTable<K, V> table;
+
+        Request(K key, V value, ForStInnerTable<K, V> table) {
+            this.key = key;
+            this.value = value;
+            this.table = table;
+        }
+
+        static <K, V> Request<K, V> of(K key, V value, ForStInnerTable<K, V> table) {
```

Comment:
A minor suggestion: 
Adding some descriptions here ? Since value is nullable and we will delete it for null value which should be explicit for callers.

Reply:
I have refined it.